pkg/controller/releasemanager/syncer.go

```diff
@@ -290,11 +421,58 @@ func (r *ResourceSyncer) garbageCollection(ctx context.Context) error {
 	return markGarbage(ctx, r.log, r.deliveryClient, r.incarnations.sorted())
 }
 
-func (r *ResourceSyncer) prepareRevisionsAndRules() ([]rmplan.Revision, []monitoringv1.Rule) {
+// returns the AlertRules from the latest deployed revision
+func (r *ResourceSyncer) prepareAlertRules() []monitoringv1.Rule {
 	alertRules := []monitoringv1.Rule{}
 
 	if len(r.incarnations.deployed()) == 0 {
-		return []rmplan.Revision{}, alertRules
+		return alertRules
+	}
+
+	alertable := r.incarnations.alertable()
+	for _, i := range alertable {
+		alertRules = i.target().AlertRules
+		break
+	}
+
+	return alertRules
+}
+
+// returns the ServiceMonitors from the latest deployed revision
+func (r *ResourceSyncer) prepareServiceMonitors() []*picchuv1alpha1.ServiceMonitor {
+	sm := []*picchuv1alpha1.ServiceMonitor{}
+
+	if len(r.incarnations.deployed()) > 0 {
+		alertable := r.incarnations.alertable()
+		for _, i := range alertable {
+			if i.target() != nil {
+				sm = i.target().ServiceMonitors
+			}
+		}
+	}
+
+	return sm
+}
+
+// returns the PrometheusRules to support SLOs from the latest deployed revision
+func (r *ResourceSyncer) prepareServiceLevelObjectives() ([]*picchuv1alpha1.ServiceLevelObjective, picchuv1alpha1.ServiceLevelObjectiveLabels) {
+	slos := []*picchuv1alpha1.ServiceLevelObjective{}
+
+	if len(r.incarnations.deployed()) > 0 {
+		releasable := r.incarnations.alertable()
```

Comment:
Will this immediately update SLOs based on latest releasable revision? Should we wait until it get's to releasing state before applying the new SLOs to protect from bad ones?

Reply:
Yeah this makes sense, thanks. I'm changing this to find the latest `releaseable()`.